### PR TITLE
Add create feedstock PR workflow

### PR DESCRIPTION
**Pull request by gsheni**
- Use a github action to instantly create a feedstock pull request when our library is release on PyPI
- Uses this workflow action: https://github.com/alteryx/create-feedstock-meta-yaml

### CODE REVIEW

File: .github/workflows/feedstock_pr_create.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          fetch-depth: '0'
+      - name: Run Create Feedstock meta YAML
+        id: create-feedstock-meta
+        uses: alteryx/create-feedstock-meta-yaml@v3
+        with:
+          project: "featuretools"
+          pypi_version: ${{ github.event.inputs.pypi_version }}
+          setup_cfg_filepath: "featuretools/setup.cfg"
+          meta_yaml_filepath: "featuretools-feedstock/recipe/meta.yaml"
+          add_to_test_requirements: "graphviz !=2.47.2"
```
Comment (thehomebrewnerd) [inline .github/workflows/feedstock_pr_create.yaml:41]: Just curious why this is needed?

File: .github/workflows/feedstock_pr_create.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+on:
+  workflow_dispatch:
```
Comment (thehomebrewnerd) [inline .github/workflows/feedstock_pr_create.yaml:2]: Is the idea here that once the pypi release is done we would manually trigger this workflow with the new version to create the feedstock PR?

**Review by thehomebrewnerd — COMMENTED** [review]
Is the intent for this to create a PR or just get everything ready and then we have to manually launch the PR with the last link that is echoed?

Also, should we create a branch in the forked repo rather than pushing the updated `meta.yaml` to `main`?

File: .github/workflows/feedstock_pr_create.yaml (hunk 2 above)
Comment (gsheni) [reply .github/workflows/feedstock_pr_create.yaml]: I have updated `release.md` with the instructions

File: .github/workflows/feedstock_pr_create.yaml (hunk 1 above)
Comment (gsheni) [reply .github/workflows/feedstock_pr_create.yaml]: What do you mean?
- conda allows you to specify `non-Python` dependencies. This is one of those.
- This was a bad release at some point (I think)
- I couldn't figure out a clean way to determine non-Python dependencies (from setup.cfg).

File: .github/workflows/feedstock_pr_create.yaml (hunk 2 above)
Comment (rwedge) [reply .github/workflows/feedstock_pr_create.yaml]: Could the `release` github action kick off this workflow after it finishes the pypi upload?

Comment (gsheni) [reply .github/workflows/feedstock_pr_create.yaml]: @rwedge What about this?
https://github.com/alteryx/featuretools/pull/2181/commits/f56f48fee51a375c46a08074185484077cff1916

File: .github/workflows/create_feedstock_pr.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        with:
+          project: "featuretools"
+          pypi_version: ${{ github.event.inputs.pypi_version }}
+          setup_cfg_filepath: "featuretools/setup.cfg"
+          meta_yaml_filepath: "featuretools-feedstock/recipe/meta.yaml"
+          add_to_test_requirements: "graphviz !=2.47.2"
+      - name: View updated meta yaml
+        run: |
+          cat featuretools-feedstock/recipe/meta.yaml
+          cd featuretools-feedstock
```
Comment (rwedge) [inline .github/workflows/create_feedstock_pr.yaml:45]:
```suggestion
        run: cat featuretools-feedstock/recipe/meta.yaml
      - name: Push updated yaml
        run: | 
          cd featuretools-feedstock
```

File: docs/source/release_notes.rst [hunk 4]
```diff
@@ -10,10 +10,11 @@ Future Release
     * Changes
     * Documentation Changes
     * Testing Changes
+        * Add create feedstock PR workflow (:pr:`2181`)
         * Limit performance tests to python 3.8 (:pr:`2198`)
-
+
```
Comment (rwedge) [inline docs/source/release_notes.rst:15]:
```suggestion
 
```

File: release.md [hunk 5]
```diff
@@ -82,6 +82,16 @@ After the release pull request has been merged into the `main` branch, it is tim
 
 ## Release on conda-forge
 
+In order to release on conda-forge, you can either wait for a bot to create a PR, or manually kickoff the creation with GitHub Actions
```
Comment (rwedge) [inline release.md:85]: After we've confirmed the release workflow can kick off the action we should update this section

Comment (gsheni) [reply release.md:85]: Agreed.

File: release.md [hunk 6]
```diff
@@ -82,6 +82,16 @@ After the release pull request has been merged into the `main` branch, it is tim
...
+
+### Option 1: Manually create the new PR with GitHub Actions
+1. Go to this GitHub Action: https://github.com/alteryx/featuretools/actions/workflows/create_feedstock_pr.yaml
+2. Input the released version with the v prefix (e.g. v0.13.3)
+3. Kickoff the GitHub action, and monitor the Job Summary. At the completion of the job, you should see summary output, with the URL: 
+  a. You can also just go to: https://github.com/machineAYX/featuretools-feedstock/pull/new/conda-autocreate-v0.13.3 (change the last part to the released version)
```
Comment (thehomebrewnerd) [inline release.md:91]: I think we might want to specify that the user also needs to visit the URL and submit the PR

File: docs/source/release_notes.rst [hunk 7]
```diff
@@ -10,10 +10,11 @@ Future Release
...
```
Comment (thehomebrewnerd) [inline docs/source/release_notes.rst:15]: Extra space

Comment (gsheni) [reply docs/source/release_notes.rst]: Fixed

File: release.md (hunk 6 above)
Comment (gsheni) [reply release.md:91]: Fixed
https://github.com/alteryx/featuretools/pull/2181/commits/854e1b1b31d8da2469eb53a3b233e7ccbcc44df3

**Review by thehomebrewnerd — APPROVED** [review]

**Pull request merged**